pyiceberg/table/__init__.py

```diff
@@ -498,7 +524,10 @@ def append(self, df: pa.Table, snapshot_properties: Dict[str, str] = EMPTY_DICT)
                     update_snapshot.append_data_file(data_file)
 
     def overwrite(
```

Comment:
I think it is worth mentioning in the docstring (probably also `api.md`) that `overwrite` operation may commit more than 1 snapshots to the table. Otherwise, user may think this will just commit an "overwrite" snapshot and have confusion when managing the snapshots later.

Reply:
That's a good point. I've updated the docstring.